src/output/mod.rs

```diff
@@ -40,8 +40,15 @@ pub async fn log(config: Config) {
 
     let mut interval = tokio::time::interval_at(start, config.general().interval());
 
-    while Instant::now() + config.general().interval() <= stop {
-        interval.tick().await;
+    while RUNNING.load(Ordering::Relaxed) && Instant::now() + config.general().interval() <= stop {
```

Comment:
I think we also need to do something similar here (https://github.com/iopsystems/rpc-perf/blob/main/src/output/mod.rs#L249) to prevent the metrics thread from blocking exits?

Reply:
Yes. Fixed.